src/main/java/com/eva/model/Model.java

```diff
@@ -115,11 +115,17 @@
     boolean hasStaffLeave(Staff target, Leave leave);
 
     /**
-     * Returns true if a staff with the same identity as {@code staff} has the same leave as {@code leave}.
-     * @return
+     * Returns the conflicting leave if a staff with the same identity as {@code staff}
+     * has the same leave period as {@code leave}.
      */
     Optional<Leave> hasLeaveDate(Staff target, LocalDate date);
 
+    /**
+     * Returns the conflicting leave if a staff with the same identity as {@code staff}
```

Comment:
Minor, but the javadoc is a little misleads readers to think the specific leave object that is conflicting will be returned.